### PR TITLE
CLOUDP-93398: Allow to reclaim free space for a cluster

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-93398

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
waiting for https://github.com/mongodb/go-client-mongodb-ops-manager/pull/114

File: internal/cli/opsmanager/clusters/clusters.go
Comment (gssbzn) [inline internal/cli/opsmanager/clusters/clusters.go:42]: [nit] just spotted this but can we move restart before Shutdown and ReclaimFreeSpaceBuilderafter StartupBuilder, seems to make sense (to me) to keep all this modification actions together, I would even move apply before create but that is a bit more fuzzy in my head

File: internal/cli/opsmanager/clusters/reclaim_free_space.go
Comment (gssbzn) [inline internal/cli/opsmanager/clusters/reclaim_free_space.go:72]: startup?

File: internal/cli/opsmanager/clusters/reclaim_free_space.go
Comment (gssbzn) [inline internal/cli/opsmanager/clusters/reclaim_free_space.go:115]: 👀

File: internal/usage/usage.go
Comment (gssbzn) [inline internal/usage/usage.go:257]: [nit] rename to prevent misuse and also a fix on copy but I would confirm with @mongodb/docs-platform 
```suggestion
	ReclaimFreeSpaceTimestamp                       = "The service reclaims the space after the current time passes the provided timestamp. If not set the current timestamp will be used."
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
One recommendation on the sorting, there are also. couple of copy/paste issues here and there

**Review by gssbzn — APPROVED** [review]
LGTM don't merge without @mongodb/docs-cloud-team LGTM

File: internal/cli/opsmanager/clusters/reclaim_free_space.go
Comment (melissamahoney-mongodb) [inline internal/cli/opsmanager/clusters/reclaim_free_space.go:89]:
```suggestion
		Short:   "Reclaim unused space for a cluster.",
```

File: internal/usage/usage.go
Comment (melissamahoney-mongodb) [inline internal/usage/usage.go:257]:
```suggestion
	ReclaimFreeSpaceTimestamp       = "The time at which the service reclaims the space. If not set, defaults to the current timestamp."
```

**Review by melissamahoney-mongodb — CHANGES_REQUESTED** [review]

File: internal/usage/usage.go
Comment (melissamahoney-mongodb) [reply internal/usage/usage.go:257]: Should we say what format the timestamp needs to be? "Timestamp in XXX format when the service reclaims the space." ?

Comment (andreaangiolillo) [reply internal/usage/usage.go]: added

File: internal/usage/usage.go
Comment (melissamahoney-mongodb) [inline internal/usage/usage.go:257]:
```suggestion
	ReclaimFreeSpaceTimestamp       = "Timestamp in ISO 8601 format when the service reclaims the space. If not set, defaults to the current timestamp."
```

**Review by melissamahoney-mongodb — COMMENTED** [review]
One tiny change then LGTM

**Review by melissamahoney-mongodb — APPROVED** [review]

**Pull request merged**
